pkg/operator/resources/operator/operator.go

```diff
@@ -1079,3 +1092,113 @@ _The CDI Operator does not support updates yet._
 		},
 	}, nil
 }
+
+func createCDIValidatingWebhook(namespace string /*, c client.Client, l logr.Logger*/) *admissionregistrationv1beta1.ValidatingWebhookConfiguration {
+	path := "/cdi-validate"
+	sideEffect := admissionregistrationv1beta1.SideEffectClassNone
+	defaultServicePort := int32(443)
+	allScopes := admissionregistrationv1beta1.AllScopes
+	exactPolicy := admissionregistrationv1beta1.Exact
+	failurePolicy := admissionregistrationv1beta1.Ignore //FIXME: should be Fail
+	defaultTimeoutSeconds := int32(30)
+	whc := &admissionregistrationv1beta1.ValidatingWebhookConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "admissionregistration.k8s.io/v1beta1",
+			Kind:       "ValidatingWebhookConfiguration",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cdi-api-validate",
+			Labels: map[string]string{
+				utils.CDILabel: apiServerServiceName, //FIXME?
+			},
+		},
+		Webhooks: []admissionregistrationv1beta1.ValidatingWebhook{
+			{
+				Name: "cdi-validate.cdi.kubevirt.io", //FIXME?
+				Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+					Operations: []admissionregistrationv1beta1.OperationType{
+						admissionregistrationv1beta1.Create,
+						admissionregistrationv1beta1.Update,
+						admissionregistrationv1beta1.Delete,
+					},
+					Rule: admissionregistrationv1beta1.Rule{
+						APIGroups: []string{cdicorev1.SchemeGroupVersion.Group},
+						APIVersions: []string{
+							cdicorev1.SchemeGroupVersion.Version,
+							cdicorev1alpha1.SchemeGroupVersion.Version,
+						},
+						Resources: []string{"cdis"},
+						Scope:     &allScopes,
+					},
+				}},
+				ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+					Service: &admissionregistrationv1beta1.ServiceReference{
+						Namespace: namespace,
+						Name:      apiServerServiceName,
```

Comment:
This should not be the same name as the apiserver service.  It should be the name of the service created in this file (which should not be the same as the apiserver service)

Reply:
Sure, but when createAPIService was added registered here with a unique name, for some reason we got 
`{"level":"info","ts":1621352271.742419,"logger":"cdi-operator","msg":"Orphan object exists","Request.Namespace":"","Request.Name":"cdi","obj":{"apiVersion":"apiregistration.k8s.io/v1beta1","kind":"APIService","name":"v1beta1.upload.cdi.kubevirt.io"}}
`